### PR TITLE
Define Populate, the inverse of Apply.

### PR DESCRIPTION
**Pull request by patrickt**
The idea and implementation are due to @jwiegley.

### CODE REVIEW

**Comment by patrickt** [comment]
Closes #25.

File: src/Data/Sum/Populate.hs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+-- | 'Populate' serves as a dispatcher for instances of 'Producible'.
+class Populate m (fs :: [Type -> Type]) where
+  populate :: m (Sum fs b)
+
+instance forall m t. (Producible m t, Alternative m) => Populate m '[t] where
+  populate = fmap inject (produce @m @t)
+
+-- We may need to unroll this instance in the future if people start using
+-- this on absolutely massive types.
```
Comment (robrix) [inline src/Data/Sum/Populate.hs:47]: I was about to ask about this 😄 Thanks for the note.

File: src/Data/Sum/Populate.hs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+-- type:
+-- @
+--   data A x = A
+--   data B x = B
+--
+--   instance Producible Parser A where produce = parseA
+--   instance Producible Parser B where produce = parseB
+--
+--   parseAlt :: Parser (Sum '[A, B] x)
+--   parseAlt = populate
```
Comment (robrix) [inline src/Data/Sum/Populate.hs:24]: I still think this is way cool.

File: src/Data/Sum/Populate.hs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+-- We may need to unroll this instance in the future if people start using
+-- this on absolutely massive types.
+instance
+  {-# OVERLAPPING #-}
+  (Populate m (u ': r), Producible m t, Alternative m) =>
+  Populate m (t ': u ': r)
+  where
+  populate =
+    fmap inject (produce @m @t)
+      <|> fmap weaken (populate @m @(u ': r))
```
Comment (robrix) [inline src/Data/Sum/Populate.hs:55]: Sick 🤘🏻 Incidentally, `<$>` has higher precedence than `<|>` so we could use it here without parens. I'm certainly not going to mandate it, however.

**Review by robrix — APPROVED** [review]
Nice ❤️

Thanks @jwiegley!

**Pull request merged**